inc/composer/class-installer.php

```diff
@@ -25,6 +25,29 @@ public function supports( $type ) {
 		return in_array( $type, [ 'wordpress-plugin', 'wordpress-muplugin' ], true );
 	}
 
+	/**
+	 * Gets all overridden packages from all extra.altis.install-overrides entries
+	 *
+	 * @return string[]
+	 */
+	protected function getAllInstallOverrides() {
```

Comment:
What's the behavior with new installs versus updates? IIRC this won't work for new installs, as at the point of `getInstallPath` being called the packages are not loaded -- that might have just been on the FS though, and I see you are reading it from the `$repo` which maybe is different.

IIRC there are three cases:

1. New install / empty vendor DIR
2. Update with modules all staying the same
3. Update with a module updating, where `$package` _could_ be the old version of the package, not the new one.

Reply:
Ah re-reading your description, it looks like `$repo->getPackages()` is _always_ the staged/updating versions in all cases?

Reply:
Correct; `$repo` is the local repository, which is the in-memory representation of Composer's state.

Running `composer update` calls our method twice per package: the first time is during a `purgePackages()` call with the existing data, and the second is from [the update itself](https://github.com/composer/composer/blob/2406a094d3b2e5d73ce105764b36898ebd19b69d/src/Composer/Installer.php#L237-L247).

Adding a `var_dump( $package->getName(), $package->getVersion() )` confirms this, with output like:

```
string(9) "altis/seo"
string(7) "4.0.0.0"
...
string(9) "altis/seo"
string(7) "5.0.0.0"
```

Since it's called multiple times, I didn't add any caching in here (which means this method gets called like a hundred times), but I suspect it would be possible to cache it if we listen to the right vars. In particular, I suspect there are changes inside `$repo` we can detect between the first call and the second.

Reply:
ok nice one, thanks for clarifying